### ISSUE TITLE
Allow props setting use case

### ISSUE THREAD

**Issue by devcrafting** [neo4j/neo4j-dotnet-driver#183]
Hello,

I was trying to set multiple properties on a node using a "props" parameter (as it is done here http://neo4j.com/docs/developer-manual/current/cypher/syntax/parameters/#cypher-parameters-create-node-with-properties).
I tried with the following anonymous parameter:

    var props = new { id = ..., name = ... }
    var parameters = new { props }
    session.Run("...", parameters)

The Run with "object" parameters does not support this use case since the ToDictionary extension (from Internal/Extensions/CollectionsExtensions.cs) transform only the first level and do not go deeper (in the "props" anonymous object).

I agree it is probably useful only in the setting props use case, but would be could great to be able to do this easily. Or we could have a Props class that manage this use case and a new Run overload?

Thanks
Clément

**Comment by zhenlineo** [comment]
Hi

You could provide multiple levels of property parameters if you use `Run(string, IDictionary<string, object>)`

```
var parameters = new IDictionary<string, object>{
    {"props", new IDictionary<string, object> { "id", ... }, {"name", ... } }
}
```
Dose this solve your problem?

Zhen

**Comment by devcrafting** [comment]
Yes that's what I did, but it is a bit more verbose than anonymous objects.
I even mixed both:

    new { props = new Dictionary<string, object> { ... } }

**Comment by zhenlineo** [comment]
https://github.com/neo4j/neo4j-dotnet-driver/pull/213

**Comment by ali-ince** [comment]
I am closing this as this is addressed by [#213](https://github.com/neo4j/neo4j-dotnet-driver/pull/213) and is already included in 1.4.1 release and merged in 1.5.0 branch.
